src/DGEList.py

```diff
@@ -223,15 +259,44 @@ def tpm(
         # TODO: Implement here
 
         # self = self.cpm(log=log, prior_count=prior_count)
-        return self
 
     def __repr__(self) -> str:
         """Give a pretty non-executeable representation of this object."""
-        num_samples = len(self.samples) if self.samples is not None else 0
-        num_genes = len(self.genes) if self.genes is not None else 0
+        num_samples = len(self._samples) if self._samples is not None else 0
+        num_genes = len(self._genes) if self._genes is not None else 0
 
         return (
             f'{self.__class__.__name__}('
             f'num_samples={num_samples:,}, '
             f'num_genes={num_genes:,})'
         )
+
+    def export_file(self, filename: str) -> None:
+        """ Convert the object to a byte representation, which can be stored or imported."""
+
+        # TODO: validate file name
+
+        print(f"Exporting data to compressed .dge file ({filename}.npz)....")
+
+        np.savez_compressed(filename,
+                            samples=self.samples,
+                            genes=self.genes,
+                            norm_factors=self.norm_factors,
+                            counts=self.counts,
+                            group=self.group)
+
+    def import_file(self, filename: str) -> None:
```

Comment:
We already have `read_handle()` for reading from a file. Refactor into one method only.